components/aws/sagemaker/batch_transform/src/sagemaker_transform_spec.py

```diff
@@ -0,0 +1,218 @@
+        ),
+        accept=InputValidator(
+            input_type=str,
+            required=False,
+            description="The MIME type used to specify the output data.",
+        ),
+        assemble_with=InputValidator(
+            input_type=str,
+            choices=["None", "Line", ""],
+            required=False,
+            description="Defines how to assemble the results of the transform job as a single S3 object. Either None or Line.",
+        ),
+        output_encryption_key=InputValidator(
+            input_type=str,
+            required=False,
+            description="The AWS KMS key that Amazon SageMaker uses to encrypt the model artifacts.",
+            default="",
+        ),
+        input_filter=InputValidator(
+            input_type=str,
+            required=False,
+            description="A JSONPath expression used to select a portion of the input data to pass to the algorithm.",
+            default="",
+        ),
+        output_filter=InputValidator(
+            input_type=str,
+            required=False,
+            description="A JSONPath expression used to select a portion of the joined dataset to save in the output file for a batch transform job.",
+            default="",
+        ),
+        join_source=InputValidator(
+            input_type=str,
+            choices=["None", "Input", ""],
+            required=False,
+            description="Specifies the source of the data to join with the transformed data.",
+            default="None",
+        ),
+        instance_type=InputValidator(
+            input_type=str,
+            required=False,
+            description="The ML compute instance type for the transform job.",
+            default="ml.m4.xlarge",
+        ),
+        instance_count=InputValidator(
+            input_type=int,
+            required=False,
+            description="The number of ML compute instances to use in the transform job.",
+        ),
+        resource_encryption_key=InputValidator(
+            input_type=str,
+            required=False,
+            description="The AWS KMS key that Amazon SageMaker uses to encrypt data on the storage volume attached to the ML compute instance(s).",
+            default="",
+        ),
+        **vars(COMMON_INPUTS),
+    )
+
+    OUTPUTS = SageMakerTransformOutputs(
+        output_location=OutputValidator(
+            description="S3 URI of the transform job results."
+        ),
+    )
+
+    def __init__(self, arguments: List[str]):
+        super().__init__(arguments, SageMakerTransformInputs, SageMakerTransformOutputs)
+
+    @property
+    def inputs(self) -> SageMakerTransformInputs:
+        return self._inputs
+
+    @property
+    def outputs(self) -> SageMakerTransformOutputs:
+        return self._outputs
+
+    @property
+    def output_paths(self) -> SageMakerTransformOutputs:
+        return self._output_paths
```

Comment:
placeholder - where is this defined

Reply:
These are defined in the super class. These getter methods are used for static type checking, so that your IDE recognises `output_paths` is of type `SageMakerTransformOutputs` rather than the default `SageMakerCommonOutputs`.